src/main/typescript/applications/prompt.ts

```diff
@@ -1,143 +1,48 @@
 import { getGame } from "../foundryHelpers.js";
+import WvI18n, { I18nRollModes } from "../wvI18n.js";
 
 /**
  * An application to prompt the user for input.
  * @typeParam Specs - the type of the input specs
  */
-export default class Prompt<Specs extends InputSpecs> extends Application {
-  /**
-   * Get data by prompting the user.
-   * @param spec - the input specification for the Prompt
-   * @param options - additional options for the Prompt
-   * @returns the user input data
-   * @typeParam I - the type of the input specs, the return type is derived off
-   */
-  static async get<I extends InputSpecs>(
-    spec: I,
-    options?: Partial<ApplicationOptions>
-  ): Promise<InputSpecsReturnType<I>> {
-    return new Promise((resolve, reject) => {
-      new this(spec, (data) => resolve(data), reject, options).render(true);
-    });
-  }
-
-  /**
-   * Get a boolean by prompting the user.
-   * @param spec - the input specification for the Prompt
-   * @param options - additional options for the Prompt
-   * @returns the boolean if resolved or an error message if rejected
-   */
-  static async getBoolean(
-    spec: Omit<CheckboxInputSpec, "type">,
-    options?: Partial<ApplicationOptions>
-  ): Promise<boolean> {
-    return new Promise((resolve, reject) => {
-      new this(
-        { value: { type: "checkbox", ...spec } },
-        (data) => resolve(data["value"]),
-        reject,
-        options
-      ).render(true);
-    });
-  }
-
-  /**
-   * Get a number by prompting the user.
-   * @param spec - the input specification for the Prompt
-   * @param options - additional options for the Prompt
-   * @returns the number if resolved or an error message if rejected
-   */
-  static async getNumber(
-    spec: Omit<NumberInputSpec, "type">,
-    options?: Partial<ApplicationOptions>
-  ): Promise<number> {
-    return new Promise((resolve, reject) => {
-      new this(
-        { value: { type: "number", ...spec } },
-        (data) => resolve(data["value"]),
-        reject,
-        options
-      ).render(true);
-    });
-  }
-
-  /**
-   * Get a string by prompting the user.
-   * @param spec - the input specification for the Prompt
-   * @param options - additional options for the Prompt
-   * @returns the string when resolved
-   */
-  static async getString(
-    spec: Omit<TextInputSpec, "type">,
-    options?: Partial<ApplicationOptions>
-  ): Promise<string> {
-    return new Promise((resolve, reject) => {
-      new this(
-        { value: { type: "text", ...spec } },
-        (data) => resolve(data["value"]),
-        reject,
-        options
-      ).render(true);
-    });
-  }
-
+export abstract class RollPrompt extends Application {
   static override get defaultOptions(): ApplicationOptions {
     const defaultOptions = super.defaultOptions;
     defaultOptions.classes.push(...[CONSTANTS.systemId, "prompt"]);
     defaultOptions.template = `${CONSTANTS.systemPath}/handlebars/prompt.hbs`;
+
```

Comment:
Why was this added? It doesn't hurt, but I also don't see any benefit. :D

Reply:
Don't remember. _Might_ have been the linter/prettier at some point, but I doubt it. Should I remove it?